src/Inventory/Request.php

```diff
@@ -319,6 +319,19 @@ public function contact($data)
         //For the moment it's the Agent who informs us about the active tasks
         $raw_data = $this->inventory->getRawData();
         if ($raw_data !== null && property_exists($raw_data, 'enabled-tasks')) {
+            $taskneededinv = [
+                'esx',
+                'netdiscovery',
+                'netinventory',
+                'remoteinventory'
+            ];
+            if (
+                !empty(array_intersect($raw_data->{'enabled-tasks'}, $taskneededinv)) &&
+                !in_array('inventory', $raw_data->{'enabled-tasks'})
+            ) {
+                $handle = $this->handleTask('inventory');
+                $response['tasks']['inventory'] = $handle;
+            }
```

Comment:
Why not just:
```suggestion
            $response['tasks']['inventory'] = $this->handleTask('inventory');
```

Reply:
> Why not just:

Because we don't need this info for all tasks, only the one in the $taskneededinv array.